crates/bevy_ecs/src/query/state.rs

```diff
@@ -1343,6 +1527,99 @@ impl fmt::Display for QueryEntityError {
     }
 }
 
+/// An error that occurs when retrieving a specific [`Entity`]'s component from a [`Query`](crate::system::Query).
+#[derive(Debug, PartialEq, Eq)]
+pub enum QueryComponentError {
```

Comment:
Should this live in `query/mod.rs` instead? I like making it pub though.

Reply:
I think it makes sense to live wherever `QueryEntityError` lives, which is currently `query/state.rs`. Moving both into `query/mod.rs` (or a separate `query/error.rs`) would be my preference.

Reply:
Let's move them both then.